src/System.Private.CoreLib/shared/System/Collections/IHashCodeProvider.cs

```diff
@@ -13,6 +14,6 @@ namespace System.Collections
     public interface IHashCodeProvider
     {
         /// <summary>Returns a hash code for the given object.</summary>
-        int GetHashCode(object obj);
+        int GetHashCode(object? obj);
```

Comment:
This allows null?

Reply:
good catch - I think I've seen the implementation which allowed null but now when I looked again there are actually no implementations here (it must have been a very specific implementation of equality comparer) - let's see if we get compile errors when we move it to other repos but for now better to keep arguments more constrained